src/providers/wms/qgswmsprovider.cpp

```diff
@@ -1130,11 +1130,20 @@ void QgsWmsProvider::createTileRequestsXYZ( const QgsWmtsTileMatrix *tm, const Q
       turl.replace( QLatin1String( "{q}" ), _tile2quadkey( tile.col, tile.row, z ) );
 
     turl.replace( QLatin1String( "{x}" ), QString::number( tile.col ), Qt::CaseInsensitive );
-    turl.replace( QLatin1String( "{y}" ), QString::number( tile.row ), Qt::CaseInsensitive );
+    // inverted Y axis
+    bool invertedY = turl.contains( QLatin1String( "{-y}" ) );
+    if ( invertedY )
+    {
+      turl.replace( QLatin1String( "{-y}" ), QString::number( tm->matrixHeight - tile.row ), Qt::CaseInsensitive );
```

Comment:
That makes a lot of sense, thanks!

Do you see any problem in backporting such a small change to 2.18? (wether this is a bugfix or a feature it is not entirely clear).